# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/ObjLiteral.cpp
# Commit: ca60758cd534
# Full Hash: ca60758cd534c0f3bb04a7acda15eb41a67e7389
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2020-11-14 09:46:25
# Regressor Bug: 1675241
# File Overlap Count: 1
# Description:
#   Bug 1675241 - Part 12: Use TaggedParserAtomIndex in ObjLiteralStencil. r=tcampbell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D96919
# ==============================================================================

diff -r 8af30c6d9bb9 -r ca60758cd534 js/src/frontend/ObjLiteral.cpp
--- a/js/src/frontend/ObjLiteral.cpp	Sat Nov 14 00:03:36 2020 +0000
+++ b/js/src/frontend/ObjLiteral.cpp	Sat Nov 14 00:03:45 2020 +0000
@@ -35,7 +35,7 @@
       return;
     case ObjLiteralOpcode::ConstAtom: {
       uint32_t index = insn.getAtomIndex();
-      JSAtom* jsatom = atoms[index]->toExistingJSAtom(cx, atomCache);
+      JSAtom* jsatom = atomCache.getExistingAtomAt(cx, atoms[index]);
       MOZ_ASSERT(jsatom);
       *valOut = StringValue(jsatom);
       return;
@@ -78,8 +78,8 @@
     if (insn.getKey().isArrayIndex()) {
       propId = INT_TO_JSID(insn.getKey().getArrayIndex());
     } else {
-      const frontend::ParserAtom* atom = atoms[insn.getKey().getAtomIndex()];
-      JSAtom* jsatom = atom->toExistingJSAtom(cx, atomCache);
+      JSAtom* jsatom =
+          atomCache.getExistingAtomAt(cx, atoms[insn.getKey().getAtomIndex()]);
       MOZ_ASSERT(jsatom);
       propId = AtomToId(jsatom);
     }
@@ -252,27 +252,25 @@
 void ObjLiteralStencil::dump() {
   js::Fprinter out(stderr);
   js::JSONPrinter json(out);
-  dump(json);
+  dump(json, nullptr);
 }
 
-void ObjLiteralStencil::dump(js::JSONPrinter& json) {
+void ObjLiteralStencil::dump(js::JSONPrinter& json,
+                             frontend::CompilationStencil* compilationStencil) {
   json.beginObject();
-  dumpFields(json);
+  dumpFields(json, compilationStencil);
   json.endObject();
 }
 
-void ObjLiteralStencil::dumpFields(js::JSONPrinter& json) {
+void ObjLiteralStencil::dumpFields(
+    js::JSONPrinter& json, frontend::CompilationStencil* compilationStencil) {
   writer_.dumpFields(json);
 
   json.beginListProperty("atoms");
   for (auto& atom : atoms_) {
-    if (atom) {
-      GenericPrinter& out = json.beginString();
-      atom->dumpCharsNoQuote(out);
-      json.endString();
-    } else {
-      json.nullValue();
-    }
+    json.beginObject();
+    frontend::DumpTaggedParserAtomIndex(json, atom, compilationStencil);
+    json.endObject();
   }
   json.endList();
 }